datafusion/functions-nested/src/range.rs

```diff
@@ -109,8 +121,11 @@ impl ScalarUDFImpl for Range {
         match args[0].data_type() {
```

Comment:
we can probably factor our this block to avoid make a modifications twice

Reply:
There is so much duplication in these UDF's. I think it's best to push the cleanup of that to a separate PR and keep this one as small as possible.